### PR TITLE
Implemented create-headless command and headless template for creating headless applications

### PR DESCRIPTION
**Pull request by jamesjhedley**
-includes template www to create cordova headless app
-includes native headless app built using momentics

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -70,6 +72,45 @@ module.exports = function CLI(args) {
     } else if (command === "create") {
         nonCLIEntry(opts);
         webworks.create.apply(this, filteredOpts);
+    } else if (command === "create-headless") {
```
Comment (bryanhiggins) [inline lib/cli.js:75]: This should probably go into its own module

File: templates/headless-template/www/config.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			type="Qnx/Elf">assets/HeadlessService</asset>
+
+		<asset entryPointId="Headless"
+			type="Qnx/Elf">wwe</asset>
+
+        <entryPoint id="Headless">
+            <entryPointType>Qnx/WebKit</entryPointType>
+        </entryPoint>
+
+		<entryPoint id="HeadlessService">
```
Comment (bryanhiggins) [inline templates/headless-template/www/config.xml]: inconsistent tabbing

File: templates/headless-template/www/config.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+<widget id="com.example.Headless" version="0.0.1" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0" xmlns:rim="http://www.blackberry.com/ns/widgets">
+    <name>Headless WebWorks Application</name>
+    <author>Your Name</author>
+    <description>A sample headless WebWorks application</description>
+    <content src="index.html" />
+    <config-file target="bar-descriptor.xml" parent="/qnx">
+		<asset entryPointId="HeadlessService"
+			type="Qnx/Elf">assets/HeadlessService</asset>
+
+		<asset entryPointId="Headless"
```
Comment (bryanhiggins) [inline templates/headless-template/www/config.xml]: Can we rename this to HeadlessUI?

File: templates/headless-template/www/index.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            <p />
+            <label for="btnResend">Headless Service says Hello!</label>
+            <button id="btnResend" type="button" onclick="resendNotification()">Resend Notification</button>
+        </div>
+        <script type="text/javascript" src="cordova.js"></script>
+        <script type="text/javascript" src="js/index.js"></script>
+        <script type="text/javascript">
+            app.initialize();
+        </script>
+        <script type="text/javascript">
```
Comment (bryanhiggins) [inline templates/headless-template/www/index.html]: Move this block into index.js

**Comment by jamesjhedley** [comment]
@bryanhiggins Addressed comments.

Currently `create-headless` behaves the same way `create` does when specifying id and name with a template. It ignores them. We can fix this in Cordova, but i personally feel like it does as it should; It should use the config.xml (if included) in the template and ignore any specified via options. In my particular scenario my headless template would break if the user was to specify a different ID, since the ID is used for the entrypoints and invoke targets. Personally i would like to update the signature to remove the id/name for the create-headless command. I will talk with you offline.

Also please note i have included my latest changes in a separate commit for reviewing purposes.
I will squash when the review/testing is complete.

**Comment by bryanhiggins** [comment]
OK, sounds good. Go ahead and remove the signature and squash.

**Comment by jamesjhedley** [comment]
Updated the signature.
Added some tests.
Implemented webworks version.
Updated MakeFile to copy headless binary to www folder IF EXISTS.
Squashed.

**Comment by bryanhiggins** [comment]
Looks good @jkeshavarzi 

I'll try this out soon.

**Comment by jamesjhedley** [comment]
FYI: I have left a prebuilt binary for HeadlessService in the template so a user can `webworks create-headless app` for instance and immediately `webworks build/run`. If they want to make changes to the native HeadlessService they can import the project (included in root of app) using Momentics and make the necessary changes. They can then call `make` and the newly built binary will be copied/overwrite the prebuilt one in www/assets. This only happens if the www/assets folder exists (Momentics - import with link). If the user decided to select the "copy into workspace" option when importing, there is nothing we can do and they will need to manually copy over the new binary. From here they call `webworks build/run` as usual.

I have not included the `webworks build` command in the `webworks create-headless` command as i feel it moves away from the standard Cordova flow(create --> build --> run). This way create-headless behaves as close as possible to `webworks create`.

The only other thing we can potentially do is update the MakeFile to `webworks build` as well but i felt this was unnecessary. 1) the user will eventually have to `webworks run` anyways and that will build the Cordova portion. 2) I think we should keep the native build process separate from the Cordova one (other than copying over the built binary into www/assets).

If you disagree with any of this, please let me know.

**Comment by bryanhiggins** [comment]
I'm good with that. My suggestion was to build cordova along with the native portion (not during create) but it's not absolutely necessary and definitely not for this pull request.

cc @kwallis

**Comment by bryanhiggins** [comment]
r+, I was able to import, build, deploy, receive notification & launch UI

**Pull request merged**